### PR TITLE
Delete legacy DataLoader processing utility

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

The function is misleading as there is no real flattening happening, we just need to convert tuple to list.

Part of #8364

### Does your PR introduce any breaking changes ? If yes, please list them.

None

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [n/a] Did you make sure to update the documentation with your changes? (if necessary)
- [n/a] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [n/a] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)
- [x] Did you list all the breaking changes introduced by this pull request?

## PR review
 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

### CODE REVIEW

File: pytorch_lightning/trainer/data_loading.py [hunk 1]
```diff
@@ -467,18 +467,8 @@ def request_dataloader(self, model: 'pl.LightningModule', stage: str) -> DataLoa
...
-        dataloader = self._flatten_dl_only(dataloader)
+        dataloader = getattr(model, f'{stage}_dataloader')()
+        if isinstance(dataloader, tuple):
+            dataloader = list(dataloader)
         self.accelerator.barrier('get_dataloaders')
         return dataloader
-
-    def _flatten_dl_only(self, dataloaders):
-        # handles user error when they return:
-        # return dl1, dl2  vs  return (dl1, dl2)
```
Comment (awaelchli) [inline pytorch_lightning/trainer/data_loading.py:477]: don't you think it was meant to handle something like `(dl1, dl2), `
not the extra comma
a function named flatten_dl would make sense in this case, even though the implementation here doesn't handle it like htat.

**Review by awaelchli — APPROVED** [review]

File: pytorch_lightning/trainer/data_loading.py (hunk 1 above)
Comment (carmocca) [reply pytorch_lightning/trainer/data_loading.py:477]: Looking at the PR that introduced it

**Review by justusschock — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGTM !
